bin/latest-stable

```diff
@@ -10,14 +34,18 @@ SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 # reject release candidates
 # reject empty line
 # finally take last one
-ASDF_ELIXIR_LATEST_VERSION=$("$SCRIPT_DIR"/list-all | tr " " "\n" | grep -Ev "^0|^main|^master|otp|rc|^$" | tail -n 1)
+ASDF_ELIXIR_LATEST_VERSION=$("$ASDF_ELIXIR_SCRIPT_DIR"/list-all | tr " " "\n" | grep -Ev "^0|^main|^master|otp|rc|^$" | tail -n 1)
 
-# gets current erlang version without headers
-# changes spaces to new line
-# takes second line (Name Version Source Installed)
-ASDF_ELIXIR_LATEST_OTP=$(asdf current --no-header erlang | tr -s " ." "\n" | sed -n 2p)
+# support for v-dev and different asdf API versions
+ASDF_ELIXIR_ASDF_VERSION="$(asdf version)"
+if [ "$ASDF_ELIXIR_ASDF_VERSION" = "v-dev" ] || version_gte_api_change "${ASDF_ELIXIR_ASDF_VERSION#v}"; then
+  # use --no-header flag only in latest asdf API (go builds)
+  ASDF_ELIXIR_LATEST_OTP=$(asdf current --no-header erlang)
+else
+  ASDF_ELIXIR_LATEST_OTP=$(asdf current erlang)
+fi
 
 # Note: asdf latest does not allows latest stable version to start with number
 #       therefore installing from source by git ref is not supported, see:
 #       https://github.com/asdf-vm/asdf/blob/v0.16.5/internal/versions/versions.go#L276
-echo "$ASDF_ELIXIR_LATEST_VERSION-otp-$ASDF_ELIXIR_LATEST_OTP"
+printf "$ASDF_ELIXIR_LATEST_VERSION-otp-%s\n" "$(printf "%s" "$ASDF_ELIXIR_LATEST_OTP" | tr -s " ." "\n" | sed -n 2p)"
```

Comment:
nitpick: No variables should be used in the first argument to `printf`. The first argument should be a format string only. Like this:

```
printf "%s-otp-%s\n" "$ASDF_ELIXIR_LATEST_VERSION" "$(printf "%s" "$ASDF_ELIXIR_LATEST_OTP" | tr -s " ." "\n" | sed -n 2p)"
```